logic/src/commonMain/kotlin/com/wire/kalium/logic/util/JsonSerializable.kt

```diff
@@ -0,0 +1,10 @@
+package com.wire.kalium.logic.util
+
+import com.wire.kalium.network.tools.KtxSerializer
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.encodeToString
+
+interface JsonSerializable {
+    fun toJson(): String = KtxSerializer.json.encodeToString(this)
+    fun fromJson(jsonString: String): Any = KtxSerializer.json.decodeFromString(jsonString)
```

Comment:
How can this be called? It would be good to have some tests to see the usage. 

I can't quite understand the way it currently is.

For `toJson` OK:

```kotlin

val originalQualifiedID = QualifiedID("value", "domain")
val myJsonString = qualifiedID.toJson()
// So far so good. Seems to work well
```


But for `fromJson`:

```kotlin

// I don't have a Qualified ID and I want to call fromJson in order to get one.
// I need an instance of QualifiedID in order to call fromJson a QualifiedID

val dumbID = QualifiedID("", "")
val originalQualifiedID = dumbID.fromJson(myJsonString)
```

Reply:
Maybe something like this can work:

```kotlin


    interface JsonSerializable{
        companion object
    }

    @Serializable
    class Foo(
        @SerialName("bar") val bar: String
    ) : JsonSerializable
    
    fun JsonSerializable.toJsonString() = KtxSerializer.json.encodeToString(this)

    inline fun <reified T : JsonSerializable> fromJsonString(jsonString: String) =
        KtxSerializer.json.decodeFromString<T>(jsonString)


    fun givenOriginalFoo_whenSerializingAndDeserializing_thenTheResultShouldMatchTheOriginal() {
        val originalFoo = Foo("someData")

        val fooString = originalFoo.toJsonString()
        val result = fromJsonString<Foo>(fooString)

        assertEquals(foo, result)
    }
```